tasks.py

```diff
@@ -0,0 +1,83 @@
+from robocorp.tasks import task
+from robocorp import browser, http, excel
+from RPA.PDF import PDF
+
+@task
+def robot_spare_bin_python():
+    """Robot to enter weekly sales data into the RobotSpareBin Industries Intranet."""
+
+    # Without this, nothing is visible as it is running headless
+    browser.configure(
+        slowmo=100,
+        browser_engine="chromium",
+        headless=False,
+    )
+    open_the_intranet_website()
+    log_in()
+    download_excel_file()
+    fill_form_with_excel_data()
+    screenshot_results()
+    export_as_pdf()
+    log_out()
```

Comment:
This whole task definition is written in a very Robot Framework style, where functions pretend to be keywords. I don't remember if the certification course is structured in a way where that makes sense, but I'm not sure if this is more or less readable like this.

The most inlined version would look something like this:
```python
@task
def submit_sales_data():
    page = browser.goto("https://robotsparebinindustries.com/")
    page.fill("#username", "maria")
    page.fill("#password", "thoushallnotpass")
    page.click("button:text('Log in')")

    http.download(url="https://robotsparebinindustries.com/SalesData.xlsx", overwrite=True)
    worksheet = excel.open_workbook("SalesData.xlsx").worksheet("data")
    for row in worksheet.as_table(header=True):
        submit_form(row)    

    page.screenshot(path="output/sales_summary.png")

    pdf = PDF()
    results = page.locator("#sales-results").inner_html()
    pdf.html_to_pdf(results, "output/sales_results.pdf")

    page.click("text=Log out")


def submit_form(row):
    page = browser.page()
    page.fill("#firstname", row["First Name"])
    page.fill("#lastname", row["Last Name"])
    page.select_option("#salestarget", str(row["Sales Target"]))
    page.fill("#salesresult", str(row["Sales"]))
    page.click("text=Submit")
```

I don't mean for you to change it to look like this, but I think it's important to think if functions that contain one line and take no arguments makes it easier to understand.

Reply:
Considering it is the first course, that is very much structured around each small step, I decided to keep the small functions for ease of understanding (in case this is first contact with python and us).

I could add a *bonus chapter* with your proposal to give ppl a taste of more real-life code.